documentation/src/main/asciidoc/topics/ref_query_vector_search.adoc

```diff
@@ -0,0 +1,115 @@
+[id='vector-search_{context}']
+= Vector search queries
+
+You can perform vector kNN searches with the Ickle query language using the special operator `+<->+` to define predicates.
+
+This is an example of kNN query:
+
+[source,sql]
+----
+from play.Item i where i.myVector <-> [7,7,7]~3
+----
+
+This query will find the items that have the `myVector` fields that are within `3` nearest neighbourhood from the vector `[7,7,7]`.
+
+Notice that in order to use this kind of search the entity, in our example `play.Item`, has to be `@Indexed` and
+the field, in our example `myVector`, should be annotated with `@Vector`.
+
+We support two kinds of vector field types:
+
+* `byte` / `Byte` (to work with byte vectors)
+* `float` / `Float` (to work with float vectors)
+
+You can have different vector fields on the same entity, but in any case you can have only one vector predicate on your queries.
+
+== Vector search parameters
+
+Both the k-value and the vector can be passed as query parameter.
+The k-value scalar can be expressed with the usual placeholder `:k` in the Ickle text.
+For the vector we can use either a placeholder for each term of the vector:
+
+[source,java]
+----
+Query<Item> query = cache.query("from play.Item i where i.floatVector <-> [:a,:b,:c]~:d");
```

Comment:
The description uses `:k` while the query has `:d`. Maybe it would be a bit clearer if `:k` is also used in the query. (same in the next example where `:b` is used)

Reply:
Sure using `k` is also more meaningful!